Marlin/src/libs/autoreport.h

```diff
@@ -23,14 +23,17 @@
 
 #include "../inc/MarlinConfig.h"
 
-template<serial_index_t AR_PORT_INDEX>
-class AutoReporter {
-public:
+template <typename Child>
+struct AutoReporter {
   millis_t next_report_ms;
   uint8_t report_interval;
+  #if HAS_MULTI_SERIAL
+    serial_index_t report_port_mask;
+    AutoReporter() : report_port_mask(SERIAL_ALL) {}
+  #endif
 
   // Override this method
-  inline void auto_report() { }
+  inline void auto_report() { static_cast<Child*>(this)->report(); }
```

Comment:
Thanks! I had meant to apply the "Curiously Recursive Template Pattern" but totally forgot this aspect. https://en.wikipedia.org/wiki/Curiously_recurring_template_pattern

Reply:
I knew it! Now it's done ;-)